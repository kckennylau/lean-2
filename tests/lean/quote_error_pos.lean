open tactic

class add_monoid (α : Type) extends has_zero α, has_add α :=
(zero_add : ∀ a : α, 0 + a = a)

lemma zero_add {α : Type} [add_monoid α] (a : α) : (0 : α) + a = a :=
add_monoid.zero_add a

meta def apply_zero_add (a : pexpr) : tactic unit :=
to_expr ``(zero_add %%a) >>= exact

example (a : nat) : 0 + a = a :=
begin
  apply_zero_add ``(tt), -- Error should be here
end

meta def apply_zero_add2 (a : pexpr) : tactic unit :=
`[apply zero_add %%a]

example (a : nat) : 0 + a = a :=
begin
  apply_zero_add2 ``(tt), -- Error should be here
end
