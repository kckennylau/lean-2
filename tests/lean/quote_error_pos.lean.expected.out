quote_error_pos.lean:14:2: error: failed to synthesize type class instance for
a : ℕ
⊢ add_monoid bool
state:
a : ℕ
⊢ 0 + a = a
quote_error_pos.lean:22:2: error: failed to synthesize type class instance for
a : ℕ
⊢ add_monoid bool
state:
a : ℕ
⊢ 0 + a = a
